sympy/physics/control/lti.py

```diff
@@ -422,6 +423,89 @@ def wrapper(*args, **kwargs):
             return func(*args, **kwargs)
     return wrapper
 
+def routh_hurwitz_table(system, first_col=False ,reduce=False):
+    """
+    Creates the Routh-Hurwitz table for the given transfer function.
+
+    Parameters
+    ==========
+    system : TransferFunction
+        The transfer function for which the Routh-Hurwitz table is to be generated.
+
+    first_col : bool, optional
+        If True, returns only the first column of the Routh-Hurwitz table.
+        Default is False.
+
+    reduce : bool, optional
+        If True, simplifies the entries in the Routh-Hurwitz table.
+        Default is False.
+
+    Returns
+    =======
+    table : Matrix
+        The Routh-Hurwitz table as a SymPy Matrix.
+
+    first_column : list, optional
+        If `first_col` is True, returns the first column of the table as a list.
+
+    Raises
+    ======
+    ValueError
+        If the given system is not a TransferFunction or has a zero denominator.
+
+    Examples
+    ========
+    >>> from sympy.physics.control import TransferFunction, routh_hurwitz_table
+    >>> from sympy.abc import s
+
+    >>> tf = TransferFunction(s**2 + 5*s + 8,1,s)
```

Comment:
I'm not sure that using `TransferFunction` like this is correct.
For transfer function, I think that you should only deal with denominator,
and you should get denominator from transfer function such as $\frac{1}{s^2+5s+8}$, not the numerator.

And as @moorepants noted, it may not reduce the generality even if we just have a function that takes `Poly` (or some other useful form of polynomial expression only of the denominator)
and not the full transfer function, because getting denominator for `TransferFunction` is very easy for users `tf.den`.

Reply:
For making routh table for Transfer functions first we need to convert that transfer function to characteristics equation by substituting the ` s ` terms in denominator to ` 0 ` .

Reply:
Sorry there was a mistake from my side .